File: app/src/main/java/info/nightscout/androidaps/plugins/PumpMedtronic/util/MedtronicConst.java
package info.nightscout.androidaps.plugins.PumpMedtronic.util;

/**
 * Created by andy on 5/12/18.
 */

public class MedtronicConst {

    static final String Prefix = "AAPS.Medtronic.";

    public class Prefs {

        public static final String PrefPrefix = "pref_medtronic_";
        public static final String PumpSerial = PrefPrefix + "serial";
        public static final String PumpType = PrefPrefix + "pump_type";
        public static final String PumpFrequency = PrefPrefix + "frequency";
        public static final String MaxBolus = PrefPrefix + "max_bolus";
        public static final String MaxBasal = PrefPrefix + "max_basal";
        public static final String FirstPumpStart = Prefix + "first_pump_use";

        public static final String LastGoodPumpCommunicationTime = Prefix + "lastGoodPumpCommunicationTime";
        public static final String LastGoodPumpFrequency = Prefix + "LastGoodPumpFrequency";

    }


}

File: pump/medtronic/src/main/java/info/nightscout/androidaps/plugins/pump/medtronic/util/MedtronicConst.kt
package info.nightscout.androidaps.plugins.pump.medtronic.util

import info.nightscout.androidaps.plugins.pump.medtronic.R

/**
 * Created by andy on 5/12/18.
 */
object MedtronicConst {

    const val Prefix = "AAPS.Medtronic."

    object Prefs {
        val PumpSerial = R.string.key_medtronic_serial
        val PumpType = R.string.key_medtronic_pump_type
        val PumpFrequency = R.string.key_medtronic_frequency
        val MaxBolus = R.string.key_medtronic_max_bolus
        val MaxBasal = R.string.key_medtronic_max_basal
        val BolusDelay = R.string.key_medtronic_bolus_delay
        val Encoding = R.string.key_medtronic_encoding
        val BatteryType = R.string.key_medtronic_battery_type
    }

    object Statistics {
        private const val StatsPrefix = "medtronic_"
        const val FirstPumpStart = Prefix + "first_pump_use"
        const val LastGoodPumpCommunicationTime = Prefix + "lastGoodPumpCommunicationTime"
        const val TBRsSet = StatsPrefix + "tbrs_set"
        const val StandardBoluses = StatsPrefix + "std_boluses_delivered"
        const val SMBBoluses = StatsPrefix + "smb_boluses_delivered"
        const val LastPumpHistoryEntry = StatsPrefix + "pump_history_entry"
        const val LastPrime = StatsPrefix + "last_sent_prime"
        const val LastRewind = StatsPrefix + "last_sent_rewind"
    }
}